### PR TITLE
PMM-6212 Package dbaas-tools

### PR DESCRIPTION
**Pull request by Sudokamikaze**
Spec: https://github.com/percona/pmm-server-packaging/pull/160

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-47ec736
client docker - perconalab/pmm-client-fb:PR-1069-47ec736
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-47ec736.tar.gz

File: .gitmodules [hunk 1]
```diff
@@ -87,6 +87,10 @@
 	path = sources/percona-images
 	url = https://github.com/Percona-Lab/percona-images
 	branch = master
+[submodule "aws-iam-authenticator"]
+	path = sources/aws-iam-authenticator/src/github.com/kubernetes-sigs/aws-iam-authenticator
+	url = https://github.com/kubernetes-sigs/aws-iam-authenticator
+	branch = v0.5.1
```
Comment (AlekSi) [inline .gitmodules:93]: We use submodules only for repositories on Percona accounts

(clickhouse_exporter is an exception, but upstream is frozen and actually refers to our fork now)

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-a84b0e6
client docker - perconalab/pmm-client-fb:PR-1069-a84b0e6
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-a84b0e6.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-41ea413
client docker - perconalab/pmm-client-fb:PR-1069-41ea413
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-41ea413.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-7e842e6
client docker - perconalab/pmm-client-fb:PR-1069-7e842e6
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-7e842e6.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-d38a3bc
client docker - perconalab/pmm-client-fb:PR-1069-d38a3bc
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-d38a3bc.tar.gz

File: build/bin/build-server-rpm [hunk 2]
```diff
@@ -231,6 +231,17 @@ build() {
                 cp -r /home/builder/clickhouse-rpm/rpmbuild/RPMS/* /home/builder/rpm/RPMS/${spec_name}-${rpm_version}
                 ls -la /home/builder/clickhouse-rpm/rpmbuild/SOURCES
                 cp -r /home/builder/clickhouse-rpm/rpmbuild/SOURCES/* /home/builder/rpm/SOURCES
+            elif [ \"${spec_name}\" == \"dbaas-tools\" ]; then
```
Comment (AlekSi) [inline build/bin/build-server-rpm:234]: Please do not add more there – we want to remove that hack, not add to it.

I think we can add GoReleaser installation to the spec itself? Or do something else, but do not add there.

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Comment by AlekSi** [comment]
No `:` in PR titles

File: build/bin/build-server-rpm (hunk 2 above)
Comment (Sudokamikaze) [reply build/bin/build-server-rpm:234]: GoReleaser is not available in repos(https://pkgs.org/search/?q=goreleaser), I had to install it by their way(documentation says to download the package from their GH release page):

```
deb/rpm:

Download the .deb or .rpm from the releases page and install with dpkg -i and rpm -i respectively.
```
https://goreleaser.com/install/

Comment (AlekSi) [reply build/bin/build-server-rpm:234]: No, `go get` is not better, do not use `go get`.

What I'm saying is that we should not do any that in that huge `if` statement hack that was added for ClickHouse and that we want to remove.

File: build/bin/build-server-rpm [hunk 3]
```diff
@@ -231,6 +231,17 @@ build() {
...
+                export PATH=\${PATH/\"/usr/local/bin:\"/}:/usr/local/bin
+
+                echo \"====> RPM build dependencies\"
+                sudo yum install -y https://github.com/goreleaser/goreleaser/releases/download/v0.142.0/goreleaser_amd64.rpm
```
Comment (AlekSi) [inline build/bin/build-server-rpm:238]: That's the only line that is different between that block and `else` block below, right? So can we move it to the spec?

Comment (Sudokamikaze) [reply build/bin/build-server-rpm:238]: Yes, you're right, but there's no way to add a URL of package into `BuildRequires:` in RPM spec file

We can add these into baseimage which is used to build RPMs inside PMM project

Comment (AlekSi) [reply build/bin/build-server-rpm:238]: I'm not opposing to adding it to the base image, but: why we can't run that command in `%build` section? Or install a binary release instead of RPM?

Comment (Sudokamikaze) [reply build/bin/build-server-rpm:238]: Because `%build` section is for build steps, not for installing dependencies, RPM handles it by `BuildRequires:`

Comment (AlekSi) [reply build/bin/build-server-rpm:238]: But _this_ place is for nothing at all :)

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-4c53a6f
client docker - perconalab/pmm-client-fb:PR-1069-4c53a6f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-4c53a6f.tar.gz

**Comment by Sudokamikaze** [comment]
@AlekSi @EvgeniyPatlan please check rpmbuild docker image, so that I can push it to registry & make a proper FB build.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-74eec6f
client docker - perconalab/pmm-client-fb:PR-1069-74eec6f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-74eec6f.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-2fd0079
client docker - perconalab/pmm-client-fb:PR-1069-2fd0079
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-2fd0079.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-614a932
client docker - perconalab/pmm-client-fb:PR-1069-614a932
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-614a932.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-614a932
client docker - perconalab/pmm-client-fb:PR-1069-614a932
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-614a932.tar.gz

**Comment by Sudokamikaze** [comment]
Rebased PR in 3315c99

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-3315c99
client docker - perconalab/pmm-client-fb:PR-1069-3315c99
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-3315c99.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1069-3315c99
client docker - perconalab/pmm-client-fb:PR-1069-3315c99
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1069-3315c99.tar.gz

File: Jenkinsfile [hunk 4]
```diff
@@ -209,6 +209,7 @@ pipeline {
                             build-server-rpm pmm-server
                             build-server-rpm pmm-update
                             build-server-rpm dbaas-controller
+                            build-server-rpm dbaas-tools
```
Comment (AlekSi) [inline Jenkinsfile:212]: This change, and the one below should be in a separate PR too – we will merge it, but we will not merge branch and submodule update

Comment (Sudokamikaze) [reply Jenkinsfile:212]: What's the sense of it?

Comment (AlekSi) [reply Jenkinsfile:212]: We will merge that line

File: .gitmodules [hunk 5]
```diff
@@ -78,7 +78,7 @@
 [submodule "pmm-server-packaging"]
 	path = sources/pmm-server-packaging
 	url = https://github.com/percona/pmm-server-packaging
-	branch = PMM-2.0
+	branch = PMM-6212-v2
```
Comment (AlekSi) [inline .gitmodules:81]: But we will not merge that line.
So, two PRs

File: build/bin/build-server [hunk 6]
```diff
@@ -21,6 +21,7 @@ ${bin_dir}/build-server-rpm percona-qan-app qan-app
 ${bin_dir}/build-server-rpm pmm-server
 ${bin_dir}/build-server-rpm pmm-update
 ${bin_dir}/build-server-rpm dbaas-controller
+${bin_dir}/build-server-rpm dbaas-tools
```
Comment (AlekSi) [inline build/bin/build-server:24]: And we will merge that line, but not "Submodule pmm-server-packaging updated" below

**Comment by AlekSi** [comment]
All that merged now
